### DISCUSSION TITLE
Old commits don't appear in `git reflog` after git push --force

### DISCUSSION THREAD

**Discussion by njohnso6** [community/community#140137] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hello,

I tried to `git push --force` a file recently. I had received a messages that I couldn't push or pull because of diverging structures. But it wasn't specific and I believed it was due to the one file I had committed on this commit. After pushing, I realized on the UI that several days worth of commits were gone in completely unrelated directories. I tried `git reflog` and `git log` and none of them have appeared. 
Interestingly, when I look at my commit history on my profile the number of them is listed from all the commits I made today, but they themselves no longer appear. Is there anything I can do?

**Comment by feijoes** [comment]
Was the person who made the lost commits you? Use `git fsck --lost-found` (or `git fsck --dangling`) to find your lost commits (dangling SHAs), and then use `git reflog` to match them with your commit messages so you can identify them.
